src/api/models.py

```diff
@@ -0,0 +1,265 @@
+                                 lazy='dynamic',
+                                 backref='participants')
+    roles = db.relationship('Role', secondary='user_role', backref='user')
+    redemption_requests = db.relationship(
+        'RedemptionRequest', backref='user', lazy='dynamic',
+        order_by='desc(RedemptionRequest.created_at)'
+    )
+
+
+class Role(Base):
+    """Models Roles to which all Andelans have."""
+
+    __tablename__ = 'roles'
+    users = db.relationship('User', secondary='user_role', backref='role')
+
+
+class Society(Base):
+    """Model Societies in Andela."""
+
+    __tablename__ = 'societies'
+    name = db.Column(db.String, nullable=False, unique=True)
+    color_scheme = db.Column(db.String)
+    logo = db.Column(db.String)
+    _total_points = db.Column(db.Integer, default=0)
+    _used_points = db.Column(db.Integer, default=0)
+
+    members = db.relationship('User', backref='society', lazy='dynamic')
+    logged_activities = db.relationship('LoggedActivity', backref='society',
+                                        lazy='dynamic')
+    cohorts = db.relationship('Cohort', backref='society', lazy='dynamic')
+    redemptions = db.relationship('RedemptionRequest', backref='society',
+                                  lazy='dynamic')
+
+    @property
+    def total_points(self):
+        """Keep track of all society points."""
+        return self._total_points
+
+    @total_points.setter
+    def total_points(self, point):
+        self._total_points += point.value
+
+    @property
+    def used_points(self):
+        """Keep track of redeemed points."""
+        return self._used_points
+
+    @used_points.setter
+    def used_points(self, redemption_request):
+        self._used_points += redemption_request.value
+
+    @property
+    def remaining_points(self):
+        """Keep track of points available for redeemption."""
+        return self.total_points - self.used_points
+
+
+class ActivityType(Base):
+    """Models activity types."""
+
+    __tablename__ = 'activity_types'
+    value = db.Column(db.Integer)
+    supports_multiple_participants = db.Column(db.Boolean, default=False)
+
+    activities = db.relationship('Activity', backref='activity_type')
+
+
+class Activity(Base):
+    """Model activities available for points."""
+
+    __tablename__ = 'activities'
+    activity_type_id = db.Column(db.String,
+                                 db.ForeignKey('activity_types.uuid'))
+    activity_date = db.Column(db.Date)
+    added_by_id = db.Column(db.String,
+                            db.ForeignKey('users.uuid'))
+
+
+class LoggedActivity(Base):
+    """Models Activities logged by fellows."""
+
+    __tablename__ = 'logged_activities'
+    value = db.Column(db.Integer, nullable=False)
+    status = db.Column(db.String, default='in review')
+    approved_at = db.Column(db.DateTime)
+    activity_date = db.Column(db.Date)
+    redeemed = db.Column(db.Boolean, nullable=False, default=False)
```

Comment:
Any one have a clue how this field will be updated? @Andretalik @georgreen @solnsumei @rnjane @zachang .
It's not updated at all in any of the current operations.

Reply:
This should be updated during the finance step of the lifecycle.